### PR TITLE
Feature/draft-js-plugin

### PR DESCRIPTION
**Pull request by edwinmp**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    "@types/core-js": "0.9.36",
+    "@types/draft-js": "^0.10.1",
+    "@types/enzyme": "^2.8.0",
+    "@types/jasmine": "^2.5.51",
+    "@types/karma": "^0.13.35",
+    "@types/react": "^15.0.27",
+    "@types/react-dom": "^15.5.0",
+    "check-dependencies": "^1.0.1",
+    "copy-webpack-plugin": "^4.0.1",
+    "core-js": "^2.4.1",
```
Comment (Andries-Smit) [inline package.json:34]: is normal dependency as it is used for the poly fill

File: package.json [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    "style-loader": "^0.18.2",
+    "ts-loader": "^2.1.0",
+    "tslint": "^5.4.3",
+    "tslint-eslint-rules": "^4.1.0",
+    "typescript": "^2.3.3",
+    "webpack": "^2.6.0",
+    "webpack-dev-server": "^2.2.0"
+  },
+  "dependencies": {
+    "babel-polyfill": "^6.23.0",
```
Comment (Andries-Smit) [inline package.json:79]: Not needed? as we use core-js for the polyfill

File: src/TextEditor.webmodeler.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    render() {
+        return createElement(TextEditor, preview.transformProps(this.props));
+    }
+
+    private static transformProps(props: TextEditorContainerProps): TextEditorProps {
+        const valueAttribute = props.stringAttribute ? props.stringAttribute.split(".")[2] : "";
+
+        return {
+            className: props.class,
+            readOnly: false,
```
Comment (Andries-Smit) [inline src/TextEditor.webmodeler.ts]: preview should always be read only

File: src/TextEditor.webmodeler.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { Component, createElement } from "react";
+import { TextEditor, TextEditorProps } from "./components/TextEditor";
+import TextEditorContainer, { TextEditorContainerProps } from "./components/TextEditorContainer";
+
+declare function require(name: string): string;
```
Comment (Andries-Smit) [inline src/TextEditor.webmodeler.ts]: Can we do this better? Move to typings? or in the tsconfig

File: src/TextEditor.xml [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8"?>
+<widget id="com.mendix.widget.custom.texteditor.TextEditor" needsEntityContext="true" offlineCapable="true" xmlns="http://www.mendix.com/widget/1.0/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.mendix.com/widget/1.0/ ../xsd/widget.xsd">
+    <name>TextEditor</name>
```
Comment (Andries-Smit) [inline src/TextEditor.xml]: should be `Rich Text`

File: src/components/TextEditor.ts [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+import Editor from "draft-js-plugins-editor";
+import createInlineToolbarPlugin, { Separator } from "draft-js-inline-toolbar-plugin";
+import createLinkifyPlugin from "draft-js-linkify-plugin";
+import { ContentState, Editor as DraftEditor, EditorState, convertFromHTML } from "draft-js";
+import { stateToHTML } from "draft-js-export-html";
+import {
+    BlockquoteButton,
+    BoldButton,
+    CodeBlockButton,
```
Comment (Andries-Smit) [inline src/components/TextEditor.ts]: remove

File: src/components/TextEditor.ts [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+    CodeButton,
```
Comment (Andries-Smit) [inline src/components/TextEditor.ts]: remove

File: package.json (hunk 1 above)
Comment (edwinmp) [reply package.json:34]: Couldn't get the widget to work in I.E using core-js. The core-js polyfils are not loaded early enough, so they're always missing in I.E. Perhaps u know how to make it work.

File: package.json (hunk 2 above)
Comment (edwinmp) [reply package.json:79]: This was the only work around to the problem in the comment before. This is what does the polyfill, not core-js. In fact, given the core-js issues I encountered here, I'd opt for replacing it entirely.

File: src/components/TextEditor.ts [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+import "draft-js-inline-toolbar-plugin/lib/plugin.css";
+import "draft-js-linkify-plugin/lib/plugin.css";
+import "../ui/TextEditor.scss";
+
+interface TextEditorProps {
+    className?: string;
+    style?: object;
+    value: string;
+    onChange?: (data: string) => void;
+    readOnly?: boolean;
```
Comment (Andries-Smit) [inline src/components/TextEditor.ts]: not optional, as you are defaultProps it

File: src/TextEditor.webmodeler.ts (hunk 3 above)
Comment (edwinmp) [reply src/TextEditor.webmodeler.ts]: Widget will be nearly invisible as readOnly (with no text), but sure. will do.

File: src/components/TextEditor.ts [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+            },
+            createElement(Editor, {
+                editorState: this.state.editorState,
+                onBlur: this.onBlur,
+                onChange: this.onChange,
+                plugins: [ this.inlineToolbarPlugin, this.linkifyPlugin ],
+                readOnly,
+                ref: this.refEditor
+            }),
+            createElement(this.inlineToolbarPlugin.InlineToolbar)
```
Comment (Andries-Smit) [inline src/components/TextEditor.ts]: Dont render the InlineToolbar on readOnly

File: src/TextEditor.webmodeler.ts (hunk 4 above)
Comment (edwinmp) [reply src/TextEditor.webmodeler.ts]: Done.

File: src/TextEditor.webmodeler.ts (hunk 3 above)
Comment (edwinmp) [reply src/TextEditor.webmodeler.ts]: Done.

File: src/TextEditor.xml (hunk 5 above)
Comment (edwinmp) [reply src/TextEditor.xml]: Done!

File: src/components/TextEditor.ts (hunk 6 above)
Comment (edwinmp) [reply src/components/TextEditor.ts]: Done!

File: src/components/TextEditor.ts (hunk 7 above)
Comment (edwinmp) [reply src/components/TextEditor.ts]: Done!

File: src/components/TextEditor.ts (hunk 8 above)
Comment (edwinmp) [reply src/components/TextEditor.ts]: Done!

File: src/components/TextEditor.ts (hunk 9 above)
Comment (edwinmp) [reply src/components/TextEditor.ts]: Done.

**Pull request closed without merging**